spec/acceptance/network_vlan_spec.rb

```diff
@@ -1,55 +1,81 @@
-    expect(result).to match(%r{ensure.*present})
-    expect(result).to match(%r{shutdown.*true})
-  end
+  describe 'handling regular VLANs' do
+    it 'creates a network VLAN' do
+      pp = <<-EOS
+      network_vlan { "44":
+        shutdown => true,
+        ensure => present,
+      }
+      EOS
+      make_site_pp(pp)
+      run_device(allow_changes: true)
+      # Are we idempotent
+      run_device(allow_changes: false)
+      # Check puppet resource
+      result = run_resource('network_vlan', '44')
+      expect(result).to match(%r{ensure.*present})
+      expect(result).to match(%r{shutdown.*true})
+    end
+
+    it 'edit a network VLAN' do
+      pp = <<-EOS
+      network_vlan { "44":
+        vlan_name => "testvlansoitis",
+        shutdown => false,
+        ensure => present,
+      }
+      EOS
+      make_site_pp(pp)
+      run_device(allow_changes: true)
+      # Are we idempotent
+      run_device(allow_changes: false)
+      # Check puppet resource after 10 seconds
+      sleep(10)
+      result = run_resource('network_vlan', '44')
+      expect(result).to match(%r{ensure.*present})
+      expect(result).to match(%r{shutdown.*false})
+      expect(result).to match(%r{vlan_name.*testvlansoitis})
+    end
 
-  it 'edit a network VLAN' do
-    pp = <<-EOS
-    network_vlan { "44":
-      vlan_name => "testvlansoitis",
-      shutdown => false,
-      ensure => present,
-    }
-    EOS
-    make_site_pp(pp)
-    run_device(allow_changes: true)
-    # Are we idempotent
-    run_device(allow_changes: false)
-    # Check puppet resource after 10 seconds
-    sleep(10)
-    result = run_resource('network_vlan', '44')
-    expect(result).to match(%r{ensure.*present})
-    expect(result).to match(%r{shutdown.*false})
-    expect(result).to match(%r{vlan_name.*testvlansoitis})
+    it 'delete a network VLAN' do
+      pp = <<-EOS
+      network_vlan { "44":
+        ensure => absent,
+      }
+      EOS
+      make_site_pp(pp)
+      run_device(allow_changes: true)
+      # Are we idempotent
+      run_device(allow_changes: false)
+      # Check puppet resource
+      result = run_resource('network_vlan', '44')
+      expect(result).to match(%r{ensure.*absent})
+    end
   end
 
-  it 'delete a network VLAN' do
-    pp = <<-EOS
-    network_vlan { "44":
-      ensure => absent,
-    }
-    EOS
-    make_site_pp(pp)
-    run_device(allow_changes: true)
-    # Are we idempotent
-    run_device(allow_changes: false)
-    # Check puppet resource
-    result = run_resource('network_vlan', '44')
-    expect(result).to match(%r{ensure.*absent})
+  describe 'handling default VLANs' do
+    it 'errors on update' do
+      pp = <<-EOS
+      network_vlan { "1":
+        vlan_name => "default",
+        shutdown => true,
+        ensure => present,
+      }
+      EOS
+      make_site_pp(pp)
+      expect(run_device(allow_changes: true, allow_errors: true).output).to include('VLAN 1 is a Cisco default VLAN and may not be updated')
```

Comment:
Receiving:

```
undefined method `output' for #<String:0x00007fcf7245f148>
```

For these two error test cases.

Reply:
fixed.